src/Agent/NewRelic/Agent/Core/Transactions/Transaction.cs

```diff
@@ -257,6 +257,26 @@ public ISegment StartMessageBrokerSegment(MethodCall methodCall, MessageBrokerDe
             return segment;
         }
 
+        public ISegment StartMessageBrokerSerializationSegment(MethodCall methodCall, MessageBrokerDestinationType destinationType, MessageBrokerAction operation, string brokerVendorName, string destinationName, string kind)
```

Comment:
@jaffinito Did we miss anything with the serialization instrumentation changes? This method and `CreateMessageBrokerSerializationSegmentData()` below are not used anywhere in this branch. 

If we don't need these methods, we can remove them (here, `ITransaction`, `NoOpTransaction`).

Reply:
I'll need to take a look and see.  I was getting the metrics in the test so I would expect it to be used.

Reply:
That is being used [here](https://github.com/newrelic/newrelic-dotnet-agent/blob/5d44c667b8d5a926a642f6bc05cae8b9e5e35760/src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/Kafka/KafkaSerializerWrapper.cs#L26).

Reply:
This is being called from [here](https://github.com/newrelic/newrelic-dotnet-agent/blob/5d44c667b8d5a926a642f6bc05cae8b9e5e35760/src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/Kafka/KafkaSerializerWrapper.cs#L29).

Reply:
Yep, I don't know why VS wouldn't find that reference when I searched for it though...